packages/tpetra/core/src/Tpetra_Details_DistributorActor.hpp

```diff
@@ -284,16 +313,15 @@ void DistributorActor::doPostsAllToAll(const DistributorPlan &plan,
                              "MPI_Alltoallv failed with error \""
                                  << Teuchos::mpiErrorCodeToString(err)
                                  << "\".");
-
-  return;
 }
 
 #if defined(HAVE_TPETRACORE_MPI_ADVANCE)
 template <class ExpView, class ImpView>
-void DistributorActor::doPostsNbrAllToAllV(const DistributorPlan &plan,
-                                           const ExpView &exports,
-                                           size_t numPackets,
-                                           const ImpView &imports) {
+void DistributorActor::doPostsNbrAllToAllVImpl(const DistributorPlan &plan,
```

Comment:
Can this be merged with `doPostsAllToAllImpl`?

Reply:
Actually, it's slightly different. For the neighborhood variant we use the number of messages that the plan gives us. For AllToAll we need to generate offsets for all ranks, even the ones that don't communicate.